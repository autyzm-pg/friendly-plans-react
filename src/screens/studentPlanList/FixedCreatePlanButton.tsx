import React, { SFC } from 'react';
import { FloatingAction } from 'react-native-floating-action';

import { Icon } from 'components';
import { palette } from 'styles';

interface Props {
  onPress: () => void;
}

export const FixedCreatePlanButton: SFC<Props> = ({ onPress }) => (
  <FloatingAction
    color={palette.primaryVariant}
    overrideWithAction
    actions={[
      {
        name: 'create-plan',
        icon: <Icon name="add" type="material" color={palette.secondary} size={32} />,
      },
    ]}
    onPressItem={onPress}
  />
);
