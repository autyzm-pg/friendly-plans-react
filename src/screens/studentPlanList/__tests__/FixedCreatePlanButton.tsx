import { shallow } from 'enzyme';
import React from 'react';
import { FixedCreatePlanButton } from '../FixedCreatePlanButton';

describe('FixedCreatePlanButton', () => {
  const props = {
    onPress: jest.fn(),
  };

  it('renders correctly', () => {
    const createPlanFloatButton = shallow(<FixedCreatePlanButton {...props} />);

    expect(createPlanFloatButton).toMatchSnapshot();
  });
});
