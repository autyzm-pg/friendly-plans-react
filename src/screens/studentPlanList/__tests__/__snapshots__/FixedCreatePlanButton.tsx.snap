// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedCreatePlanButton renders correctly 1`] = `
<FloatingAction
  actions={
    Array [
      Object {
        "icon": <Icon
          color="#FE9DCA"
          name="add"
          size={32}
          type="material"
        />,
        "name": "create-plan",
      },
    ]
  }
  actionsPaddingTopBottom={8}
  animated={true}
  buttonSize={56}
  color="#4F3DD5"
  dismissKeyboardOnPress={false}
  distanceToEdge={30}
  iconColor="#fff"
  iconHeight={15}
  iconWidth={15}
  listenKeyboard={false}
  mainVerticalDistance={0}
  onPressItem={[MockFunction]}
  openOnMount={false}
  overlayColor="rgba(68, 68, 68, 0.6)"
  overrideWithAction={true}
  position="right"
  shadow={Object {}}
  showBackground={true}
  visible={true}
/>
`;
